### ISSUE TITLE
Authenthication failed with key authentication in paramiko 2.9+

### ISSUE THREAD

**Issue by mawilms** [paramiko/paramiko#2143]
Hello everyone,

I've created a ssh key pair via `ssh-keygen -f bla -C ""` and I'm able to log into the customers sftp server via the command line.

Unfortunately it is not possible to use the same private key to log into the server via paramiko unless I use Paramiko 2.8.1. I've used 2.12.0 before. Interesting is that I was able to establish a ssh connection via paramiko 2.12.0 to a sftp server run via the SFTP Docker image by atmoz.

Could be related to [1135](https://github.com/paramiko/paramiko/issues/1135) and [1791](https://github.com/paramiko/paramiko/issues/1791)

Tl;dr:

- [x] Command line ssh and sftp connection successful
- [x] SFTP connection via paramiko 2.12.0 to Atmoz SFTP Docker image
- [x] SFTP connection via paramiko 2.8.1 to customer server
- [ ] SFTP connection via paramiko 2.12.0 to customer server

**Comment by avatarofhope2** [comment]
I ran into this problem with 2.12.0

Once I added `disabled_algorithms={"pubkeys": ["rsa-sha2-256", "rsa-sha2-512"]}`, it worked once again, but why are the algorithms a list if it will never get to the algorithm that actually works? I don't think I understand the root cause here. Is there something else going on here? is the issue here how the key was generated?

**Comment by upperdeck** [comment]
I am struggling with this same type ssh issue.. when i add the code disabled_algorithms={"pubkeys": ["rsa-sha2-256", "rsa-sha2-512"]}  I then get 

Exception: (<class 'TypeError'>, TypeError("'NoneType' object does not support the context manager protocol"), <traceback object at 0x0315D9A8>)

the key_path is pointing to a PEM file
thoughts? 

All is working fine on this server using ppk file and filezilla.. and the code works fine on our original server, but the new one has a newer version of paramiko.  So I added the disabled_algorithims code

code is

ssh.connect(hostname = 'xxx.xx.xxx.xxx', 
		port=22, 
	        username="username", 
	        key_filename=key_path, 
	        password = pwd,
        	disabled_algorithms={"pubkeys": ["rsa-sha2-256", "rsa-sha2-512"]})

**Comment by bskinn** [comment]
Pinging this as relevant to the key/auth overhaul of #387.
